tests/integration/test_charm.py

```diff
@@ -81,8 +82,37 @@ async def test_health_check_kfam(ops_test):
     assert result.status_code == 200
 
 
+async def test_create_profile_action_no_poddefaults(lightkube_client, ops_test):
+    """
+    Test failure of create-profile action.
+
+    PodDefault CRD doesn't exist if admission-webhook charm is not deployed causing apply
+    PodDefaults to fail.
+    """
+    username = "admin"
+    profile_name = "profilex"
+    action = (
+        await ops_test.model.applications[CHARM_NAME]
+        .units[0]
+        .run_action(
+            "create-profile",
+            username=username,
+            profilename=profile_name,
+        )
+    )
+    action_result = await action.wait()
+    assert action_result.status == "failed"
+
+
 async def test_create_profile_action(lightkube_client, ops_test):
-    """Test profile creation action."""
+    """
+    Test profile creation action.
+
+    Deploy admission-webhook before testing success of the actions to enable applying PodDefaults,
+    PodDefault CRD is part of admission-webhooks's CRDs.
+    """
+    await ops_test.model.deploy(ADMISSION_WEBHOOK_NAME, channel="latest/edge")
+    await ops_test.model.wait_for_idle(apps=[ADMISSION_WEBHOOK_NAME], status="active")
```

Comment:
I can't decide what to think about this.  It feels a bit strange to deploy some charm dependencies during a test (I'd expect a `test_create_profile_action` to be idempotent and something I could run twice in a row), but then I can't think of a better way.  Maybe we can add a comment in the docstring highlighting how this is not idempotent and will not pass if executed twice in the same cluster?  But then again, so are all our tests that deploy things...

I'm ok with this as is.  Just figured I'd raise it in case you saw a better alternative

Reply:
@ca-scribner I think this deserves its own conversation. Let's discuss it and decide, for now, as you said, this is okay.